src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +13,400 @@
+        
+        
+        fsForm.setInstructions(sanitizeForHtml(newFeedbackSession.instructions.getValue()));
+        
+        fsForm.setFsStartDate(TimeHelper.formatDate(newFeedbackSession.startTime));
+        
+        Date date;
+        date = newFeedbackSession.startTime;
+        fsForm.setFsStartTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        fsForm.setFsEndDate(TimeHelper.formatDate(newFeedbackSession.endTime));
+        
+        
+        date = newFeedbackSession.endTime;
+        fsForm.setFsEndTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        fsForm.setGracePeriodOptions(getGracePeriodOptionsAsElementTags(newFeedbackSession.gracePeriod));
+        
+        fsForm.setSubmitButtonDisabled(false);
+        fsForm.setFormSubmitAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_EDIT_SAVE));
+        fsForm.setSubmitButtonText(SAVE_CHANGES);
+        fsForm.setSubmitButtonVisible(false);
+        
+        fsForm.setEditButtonTags(new ElementTag("onclick", "enableEditFS()"));
+    }
+
+    private AdditionalSettingsFormSegment buildFsFormAdditionalSettings(FeedbackSessionAttributes newFeedbackSession) {
+        
+        Date date;
+        AdditionalSettingsFormSegment additionalSettings = new AdditionalSettingsFormSegment(); 
+        boolean hasSessionVisibleDate = !TimeHelper.isSpecialTime(newFeedbackSession.sessionVisibleFromTime);
+        additionalSettings.setSessionVisibleDateButtonChecked(hasSessionVisibleDate);
+        additionalSettings.setSessionVisibleDateValue(hasSessionVisibleDate ? 
+                                                      TimeHelper.formatDate(newFeedbackSession.sessionVisibleFromTime) :
+                                                      "");
+        additionalSettings.setSessionVisibleDateDisabled(!hasSessionVisibleDate);
+        
+        date = hasSessionVisibleDate ? newFeedbackSession.sessionVisibleFromTime : null;   
+        additionalSettings.setSessionVisibleTimeOptions(getTimeOptionsAsElementTags(date));
+        
+        additionalSettings.setSessionVisibleAtOpenChecked(Const.TIME_REPRESENTS_FOLLOW_OPENING.equals(
+                                                                   newFeedbackSession.sessionVisibleFromTime));
+        
+        additionalSettings.setSessionVisiblePrivateChecked(Const.TIME_REPRESENTS_NEVER.equals(
+                                                               newFeedbackSession.sessionVisibleFromTime));
+                        
+        boolean hasResultVisibleDate = !TimeHelper.isSpecialTime(newFeedbackSession.resultsVisibleFromTime);
+        
+        additionalSettings.setResponseVisibleDateChecked(hasResultVisibleDate);
+        
+        additionalSettings.setResponseVisibleDateValue(hasResultVisibleDate ? 
+                                                       TimeHelper.formatDate(newFeedbackSession.resultsVisibleFromTime) :
+                                                       "");
```

Comment:
If you intend to use methods to set the variables, it may be tidier to simply use 2 methods for the Session and Response (e.g. `setSessionOptions()`).